scripts/oneoff/reset-supplier-declaration.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+"""
+A one-off script for resetting a supplier's declarations after a temporary bug affected the G-Cloud 10 application
+process, preventing suppliers from copying over answers from their G-Cloud 9 declaration.
+
+You can reset a declaration either by providing the email address of a user associated with that supplier, or by
+directly using the supplier's id.
+
+Syntax: ./scripts/oneoff/reset-supplier-declaration.py --stage preview --email 12345@user.marketplace.team
+        ./scripts/oneoff/reset-supplier-declaration.py --stage preview --supplier-id 123456
+"""
+
+import getpass
+import sys
+
+import argparse
+
+sys.path.insert(0, '.')  # noqa
+
+from dmapiclient import DataAPIClient
+from dmapiclient.errors import HTTPError
+from dmscripts.helpers.auth_helpers import get_auth_token, get_api_url
+
+
+def reset_supplier_declaration(stage, framework_slug, reason, email, supplier_id):
+    data_api_token = get_auth_token('api', stage) if stage != 'development' else 'myToken'
+    data_api_client = DataAPIClient(get_api_url('api', stage), data_api_token)
+
+    if email:
+        user = data_api_client.get_user(email_address=email)
+
+        if not user:
+            print(f'No user found for email address `{email_address}`')
+            exit(1)
+
+        user_supplier_id = user['users']['supplier']['supplierId']
+        if user_supplier_id and supplier_id and user_supplier_id != supplier_id:
+            print(f'Email address provided does not match with supplier provided. Email address `{email}` is '
+                  f'associated with supplierId `{supplier_id}`. Script was called with supplierId `{supplier_id}`.')
+            exit(2)
+
+    try:
+        supplier_framework = data_api_client.get_supplier_framework_info(supplier_id=supplier_id,
```

Comment:
If they pass in an email but not a supplier_id, this will fail. You need to set `supplier_id`